sdk/instrumentation/instrumentation-opentelemetry/src/transformations.ts

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  InstrumenterSpanOptions,
+  TracingSpanContext,
+  TracingSpanKind,
+  TracingSpanLink
+} from "@azure/core-tracing";
+import {
+  Link,
+  SpanAttributeValue,
+  SpanAttributes,
+  SpanContext,
+  SpanKind,
+  SpanOptions,
+  TraceFlags,
+  TraceState
+} from "@opentelemetry/api";
+
+import { TRACEPARENT_HEADER_VERSION } from "./constants";
+
+/**
+ * Converts our TracingSpanKind to the corresponding OpenTelemetry SpanKind.
+ *
+ * By default it will return {@link SpanKind.INTERNAL}
+ * @param tracingSpanKind - The core tracing {@link TracingSpanKind}
+ * @returns - The OpenTelemetry {@link SpanKind}
+ */
+export function toOpenTelemetrySpanKind<K extends TracingSpanKind>(
+  tracingSpanKind?: K
+): SpanKindMapping[K] {
```

Comment:
some fancy TS in here 😄

Reply:
You probably already figured this out, but this is courtesy of Will - it's way above my paygrade 😄